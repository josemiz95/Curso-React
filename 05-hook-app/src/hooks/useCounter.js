import { useState } from "react"

export const useCounter = (initialState = 1) => {
    const [state, setState] = useState(initialState);

    const operate = num => setState(state+num);
    const reset = num => setState(initialState);

    const increment = (factor=1) => operate(factor);
    const decrement = (factor=1) => operate(-1*factor);

    return {state, increment, decrement, operate, reset};
}
