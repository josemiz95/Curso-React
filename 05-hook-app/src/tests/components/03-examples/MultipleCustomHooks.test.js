import {shallow}  from "enzyme"
import {MultipleCustomHooks}  from "../../../components/03-examples/MultipleCustomHooks";
import {useFetch}  from "../../../hooks/useFetch";

jest.mock('../../../hooks/useFetch');

describe('Test on <MultipleCustomHooks />', () => { 
	test('should be shown correctly', () => { 
		useFetch.mockReturnValue({ data: null, loading: true, error: null });

		const wrapper = shallow(<MultipleCustomHooks />);

		expect(wrapper).toMatchSnapshot();
	});

	test('should to show the info', () => { 
		useFetch.mockReturnValue({
			data: [{
				author: 'Josemi',
				quote: 'Hola Mundo'
			}],
			loading: false,
			error: null
		});

		const wrapper = shallow(<MultipleCustomHooks />);
		expect(wrapper).toMatchSnapshot();
		expect(wrapper.find('.quote').text().trim()).toBe('Hola Mundo');
		expect(wrapper.find('.blockquote-footer').text().trim()).toBe('Josemi');

	});
});