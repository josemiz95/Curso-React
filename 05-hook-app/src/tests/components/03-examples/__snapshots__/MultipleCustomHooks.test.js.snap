// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <MultipleCustomHooks /> should be shown correctly 1`] = `
<div>
  <h1>
    BreakingBad Quotes
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    Loading ...
  </div>
  <button
    className="btn btn-primary"
    disabled={true}
    onClick={[Function]}
  >
    Anterior
  </button>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Siguiente
  </button>
</div>
`;

exports[`Test on <MultipleCustomHooks /> should to show the info 1`] = `
<div>
  <h1>
    BreakingBad Quotes
  </h1>
  <hr />
  <blockquote
    className="blockquote text-end"
  >
    <p
      className="quote"
    >
      Hola Mundo
    </p>
    <footer
      className="blockquote-footer"
    >
       
      Josemi
       
    </footer>
  </blockquote>
  <button
    className="btn btn-primary"
    disabled={true}
    onClick={[Function]}
  >
    Anterior
  </button>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Siguiente
  </button>
</div>
`;
